### PR TITLE
SPML/UCX: fixed hang in SHMEM_FINALIZE

### PR DESCRIPTION
**Pull request by hoopoepg**
- used MPI _Barrier to synchronize processes

### CODE REVIEW

File: oshmem/mca/spml/ucx/spml_ucx.c [hunk 1]
```diff
@@ -124,18 +124,16 @@ int mca_spml_ucx_del_procs(ompi_proc_t** procs, size_t nprocs)
...
     }
 
-    ret = opal_common_ucx_del_procs(del_procs, nprocs, oshmem_my_proc_id(),
-                                    mca_spml_ucx.num_disconnect,
-                                    mca_spml_ucx_ctx_default.ucp_worker);
-
+    ret = opal_common_ucx_del_procs_nofence(del_procs, nprocs, oshmem_my_proc_id(),
+                                            mca_spml_ucx.num_disconnect,
+                                            mca_spml_ucx_ctx_default.ucp_worker);
+    /* Do not barrier here - barrier is called in _shmem_finalize */
```
Comment (yosefe) [inline oshmem/mca/spml/ucx/spml_ucx.c]: No need to barrier here - barrier is called in _shmem_finalize

File: oshmem/mca/spml/ucx/spml_ucx.c [hunk 2]
```diff
@@ -323,6 +321,8 @@ int mca_spml_ucx_add_procs(ompi_proc_t** procs, size_t nprocs)
     free(wk_roffs);
 
     SPML_UCX_VERBOSE(50, "*** ADDED PROCS ***");
+
+    opal_common_ucx_mca_proc_added();
```
Comment (yosefe) [inline oshmem/mca/spml/ucx/spml_ucx.c]: why is this needed?

File: oshmem/mca/spml/ucx/spml_ucx.c (hunk 1 above)
Comment (hoopoepg) [reply oshmem/mca/spml/ucx/spml_ucx.c]: fixed

File: oshmem/mca/spml/ucx/spml_ucx.c (hunk 2 above)
Comment (hoopoepg) [reply oshmem/mca/spml/ucx/spml_ucx.c]: there is detection of missing event handler. original call was from incorrect function

**Comment by yosefe** [comment]
bot:retest

**Comment by hoopoepg** [comment]
bot:retest

**Comment by hoopoepg** [comment]
@yosefe ok to merge?

**Comment by yosefe** [comment]
@hoopoepg pls port to 4.0.x

**Comment by jsquyres** [comment]
This PR should not have been merged as it was -- the commit message message on the first commit directly contradicts the commit content.

Mistakes happen, but this particular pattern "make a commit, and then make another commit to fix the first commit" is a bit of a pet peeve of mine.  Please make the final merged commits be good, high-quality commits whenever possible.  The two commits on this PR should have been squashed before merging so that we didn't have to have a bad commit on master in the first place (which then gets propagated out to the release branches).  Thanks.

**Comment by gpaulsen** [comment]
Oh, sorry I should have caught that in review.
